rainbow-me/rainbowkit
# chore: pnpm version 9, dependencies upgrades and esbuild 20.x

## Changes
- Upgraded following dependencies:
  - `contentlayer` / `next-contentlayer` to `0.3.4` - (site -> package.json)
  - `@vanilla-extract/css` to `1.15.1` - (site -> package.json)
  - `@vanilla-extract/next-plugin` to `2.4.0` - (site -> package.json)
  - `@vanilla-extract/recipes` to `^0.5.2` - (site -> package.json)
  - `@vanilla-extract/esbuild-plugin` to `^2.3.5` - (package.json)
  - `@vanilla-extract/vite-plugin` to `^4.0.9` - (package.json)
  - `esbuild` to `^0.20.2` - (package.json)
- Refactored build files for each package in `/packages/*` folder to include the newest `esbuild` features
- Updated CI/CD to use the newest pnpm version 9


## What to test
- Make sure you can run `pnpm install` and `pnpm run dev` across all packages / examples without having any crashes
- Make sure to test the RainbowKit modal wallet icons to see if `esbuild` bundles them correctly as it did in `0.14.39` version
